pkg/cmd/roachtest/tpcc.go

```diff
@@ -728,6 +728,14 @@ func runTPCCBench(ctx context.Context, t *test, c *cluster, b tpccBenchSpec) {
 				t.Fatal(err)
 			}
 			c.Run(ctx, loadNodes, "./cockroach gen haproxy --insecure --url {pgurl:1}")
+			// Increase the maximum connection limit to ensure that no TPC-C
+			// load gen workers get stuck during connection initialization.
+			// 10k warehouses requires at least 20,000 connections, so add a
+			// bit of breathing room and check the warehouse count.
+			c.Run(ctx, loadNodes, "sed -i 's/maxconn 4096/maxconn 21000/' haproxy.cfg")
```

Comment:
Should we increase the limit in `gen haproxy` instead?

The sed command will fail silently if we ever change the default; we should either make it match on `maxconn [0-9]+` or check the number of changes made.

Reply:
> Should we increase the limit in gen haproxy instead?

I'd rather not make an external change like that just to fix a test.

> we should either make it match on maxconn [0-9]+

Done.